src/java_bytecode/library/src/java/lang/Object.java

```diff
@@ -0,0 +1,135 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.lang;
+
+import org.cprover.CProver;
+import java.lang.NullPointerException;
+import java.lang.IllegalMonitorStateException;
+
+public class Object {
+
+    // lock needed for synchronization in cbmc
+    // used by monitorenter, monitorexit, wait, and notify
+    // Not present in the original Object class
+    public int monitorCount;
+    
+    public Object() {
+      monitorCount = 0;
+    }
+    
+    public final Class<?> getClass() {
+      /*
+       * MODELS LIBRARY {
+       *   We make this call to Class.forName to ensure it is loaded
+       *   by CBMC even with --lazy-methods on. We have to do this
+       *   because the internal support for getClass use the model of
+       *   Class.forName.
+       * }
+       */
+      Class c = Class.forName("");
+      return CProver.nondetWithoutNull();
+    }
+
+    public int hashCode() {
+      return 0;
+    }
+
+    public boolean equals(Object obj) {
+        return (this == obj);
+    }
+
+    protected Object clone() throws CloneNotSupportedException {
+      throw new CloneNotSupportedException();
+    }
+
+    public String toString() {
+        return getClass().getName() + "@" + Integer.toHexString(hashCode());
+    }
+
+    public final void notify()
+    {
+      // FIXME: the thread must own the lock when it calls notify
+    }
+
+    // See implementation of notify
+    public final void notifyAll()
+    {
+      // FIXME: the thread must own the lock when it calls notifyAll
+    }
+
+    public final void wait(long timeout) throws InterruptedException {
+      // FIXME: the thread must own the lock when it calls wait
+      // FIXME: should only throw if the interrupted flag in Thread is enabled
+      throw new InterruptedException();
+    }
+
+    public final void wait(long timeout, int nanos) throws InterruptedException
+    {
+        if (timeout < 0)
+        {
+            throw new IllegalArgumentException("timeout value is negative");
+        }
+
+        if (nanos < 0 || nanos > 999999) {
+          throw new IllegalArgumentException(
+            "nanosecond timeout value out of range");
+        }
+
+        if (nanos > 0) {
+          timeout++;
+        }
+
+        wait(timeout);
+    }
+
+    public final void wait() throws InterruptedException {
+        wait(0);
+    }
+
+    protected void finalize() throws Throwable { }
+    
+    public static void monitorenter(Object object)
```

Comment:
Document this new method

Reply:
done

Reply:
Use camelCase. Also as suggested above, prefix your own methods with `cprover` to distinguish them from original JDK methods.

Reply:
@dcattaruzza We will change this to `cproverMonitorEnter`, following the naming scheme that the model library takes on other methods. Please update also the documentation